bin/dune_init.ml

```diff
@@ -195,6 +195,64 @@ module Init_context = struct
   ;;
 end
 
+module Public_name = struct
```

Comment:
Is this not the same as `Lib_name`?

Reply:
not quite since it's using `Opam_compatible` and there's no `Private` version. but I'd like to make that a bit closer. maybe I can use the Lib_name parser and validate after the fact.

Reply:
I did that, I think it looks better indeed.